sycl/include/sycl/reduction.hpp

```diff
@@ -1197,15 +1356,18 @@ template <> struct NDRangeReduction<reduction::strategy::range_basic> {
         Redu.getReadWriteAccessorToInitializedGroupsCounter(CGH);
     local_accessor<int, 1> DoReducePartialSumsInLastWG{1, CGH};
 
-    auto Identity = Redu.getIdentity();
+    auto IdentityContainer = Redu.getIdentityContainer();
     auto BOp = Redu.getBinaryOperation();
 
-    using Name = __sycl_reduction_kernel<reduction::MainKrn, KernelName,
-                                         reduction::strategy::range_basic>;
+    using Name =
+        __sycl_reduction_kernel<reduction::MainKrn, KernelName,
+                                reduction::strategy::range_basic,
+                                std::bool_constant<Reduction::has_identity>>;
```

Comment:
I'm not sure we actually need the `has_identity` bit here. I'd expect all the dispatch logic to be done via `if constexpr` so we'd end up with a single scenario anyway.

Reply:
After a brief offline discussion, I agree that we don't necessarily need this, unless there is some edge-case we are missing where it would be reasonable to expect the same name for paths with identity and without.